thunder/dynamo/utils.py

```diff
@@ -462,6 +462,8 @@ def _get_example_input_tensor_metadata(t: torch.Tensor) -> ExampleInputMetaData:
     min_val = None
     max_val = None
     if not isinstance(t, FakeTensor) and t.device.type != "meta" and t.numel() != 0:
+        if t.dtype in (torch.float8_e4m3fn, torch.float8_e4m3fnuz, torch.float8_e5m2, torch.float8_e5m2fnuz):
```

Comment:
While it's OK to upcast `t` so that the min and maximum values can be computed, won't this change the dtype recorded for `t`?

Maybe an `aminmax` helper would be useful?